source/plugins/ruby/out_mdm.rb

```diff
@@ -119,6 +119,10 @@ def start
             if @isAADMSIAuth && !@isWindows
               @log.info "using IMDS sidecar endpoint for MSI token since its Arc k8s and Linux node"
               @useMsi = true
+              custom_resource_endpoint = ENV["customResourceEndpoint"]
+              if !custom_resource_endpoint.nil? && !custom_resource_endpoint.empty?
+                @@token_resource_audience = custom_resource_endpoint.strip
```

Comment:
what's the custom token resource audience look like? can you please share the audience names?

Reply:
For now, we suppose it to be in the format `https://monitor.{FQDN}/` while no final decision has been done yet.

Reply:
> For now, we suppose it to be in the format `https://monitor.{FQDN}/` while no final decision has been done yet.

> Did you test this change for both metrics and LA ingestion in Managed Identity mode?

Reply:
This PR is only intended to support Metrics in MSI mode. Will make changes for Logs in a separate PR later.

Reply:
> This PR is only intended to support Metrics in MSI mode. Will make changes for Logs in a separate PR later.

> @ms-hujia , are you using the Arc K8s addon-token adapter which we have as side car container? If yes, then audience needs to be updated in the CRD and same audience should be used in ruby code.